### PR TITLE
BLD: `ctest` port

### PR DESCRIPTION
**Pull request by mtsokol**
`ctest` port

### CODE REVIEW

**Comment by HaoZeke** [comment]
@mtsokol I updated these tests to ensure real failures are propagated correctly, similar changes are likely required in the other `test` PRs.

File: ctest/c_cblat1.f
Comment (HaoZeke) [inline ctest/c_cblat1.f:1]: `error stop` is the more natural Fortran method, but it depends on Fortran 2008 which seems like a big ask for OpenBLAS, `call abort` [1] is very well supported.

[1] https://docs.oracle.com/cd/E19957-01/805-4942/6j4m3r8sg/index.html

**Comment by rgommers** [comment]
Question: are changes to Fortran files upstreamable? In my experience it's going to be significantly easier to maintain/update if there's only changes specific to the Meson build that are parallel to the Makefile build, and no source-level changes.

**Comment by HaoZeke** [comment]
> Question: are changes to Fortran files upstreamable? In my experience it's going to be significantly easier to maintain/update if there's only changes specific to the Meson build that are parallel to the Makefile build, and no source-level changes.

Yup, it basically switches from `stop` to `call abort`, which are equally well supported (in terms of language standards / compilation). The `stop` form basically reported no errors (i.e. return code was always `0`), and didn't include a backtrace.

**Pull request merged**